### PR TITLE
Closes #1938: Prepare Quickstart for Drupal 9.5.x (and 10.x).

### LINKED ISSUE #1938: Prepare for Drupal 9.5.x
## Motivation
Drupal 9.5.0 is planned for December 2022.  9.5.0-beta1 was released September 16.

## Proposed Resolution
We need to ensure Quickstart is ready to be updated with 9.5.x as part of a future minor release.

### PR DESCRIPTION
**Pull request by joeparsons**
## Description

Drupal 9.5.0 will be released this month.  We'd like to update Quickstart to use this new Drupal core minor version as part of our next Quickstart minor release (`2.6.0`) if possible.

We won't want to merge this until the final Drupal 9.5.0 release is available.

### 9.5.x / 10.x specific changes
- Updated Drupal core patches that no longer applied with Drupal core `9.5.x`
- Addressed PHP coding standards issues flagged by updated PHPCS configuration
- Updated JS transpilation dependencies in package.json to match Drupal `9.5.x`
- Re-generated `.js` files from `.es6.js` files using updated JS tools
- Updated version of NodeJS installed by ProboCI to `18.x` (current LTS release)
- Added Drupal 10 to `core_version_requirement` property in `.info.yml` files
- Removed deprecated core modules from list of modules to install by default (color and RDF)
- Added DB update for uninstalling depreecated core modules (color, hal, quickedit, rdf).  Ckeditor (4) and aggregator are also deprecated

### Changes that might need a separate PR so they can be backported
- ~Added steps to `.probo.yml` to install the Upgrade Status module on Probo review sites~ (moved to #2135)
- ~Addressed Twig deprecations (see https://twig.symfony.com/doc/2.x/deprecated.html#tags)~ (moved to #2107)
  > ~Adding an `if` condition on a `for` tag is deprecated in Twig 2.10. Use a `filter` filter or an "if" condition inside the "for" body instead (if your condition depends on a variable updated inside the loop).~ 	
- ~Added explicit access checks to entity queries for https://www.drupal.org/node/3201242 (deprecation introduced in Drupal 9.2.x)~ (moved to #2107)

## Related issues
Closes #1938 

## How to test
General testing to ensure that Quickstart features work as expected and checking to see that no errors show up in the logs or status report.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [x] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [x] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -138,13 +138,13 @@
...
                 "Ajax CSS load order issue (1461322)": "https://www.drupal.org/files/issues/2020-06-05/1461322-25.patch",
                 "Layout builder revisions issue (3033516)": "https://www.drupal.org/files/issues/2019-06-17/3033516-17.patch",
-                "Allow text field to enforce a specific text format (784672)": "https://www.drupal.org/files/issues/2022-05-13/784672-231.patch",
+                "Allow text field to enforce a specific text format (784672)": "https://www.drupal.org/files/issues/2022-10-21/784672-271.patch",
                 "Unable to select single view display for view settings (2552541)": "https://www.drupal.org/files/issues/2020-11-03/2552541-43.patch",
-                "MediaEmbed conflates default view mode with user-selected view mode (3109289)": "https://www.drupal.org/files/issues/2020-11-11/drupal_core-mediaembed_default_view_mode-3109289-24.patch",
+                "MediaEmbed conflates default view mode with user-selected view mode (3109289)": "https://gist.githubusercontent.com/trackleft/639c796fd446db4a10988a4d90a225c9/raw/344b57b20be022d2ca0c8445eaaf3861b72a9fc5/drupal_core-mediaembed_default_view_mode-3109289-65.patch",
```
Comment (trackleft) [inline composer.json:143]:
```suggestion
                "MediaEmbed conflates default view mode with user-selected view mode (3109289)": "https://www.drupal.org/files/issues/2022-12-07/drupal_core-mediaembed_default_view_mode-3109289-65.patch",
```

File: modules/custom/az_paragraphs/az_paragraphs.install [hunk 2]
```diff
@@ -15,6 +15,7 @@ function az_paragraphs_update_920301(&$sandbox) {
 
   if (!isset($sandbox['total'])) {
     $count = \Drupal::entityQuery('paragraph')
+      ->accessCheck(FALSE)
```
Comment (trackleft) [inline modules/custom/az_paragraphs/az_paragraphs.install:18]: Was this on purpose?

File: modules/custom/az_paragraphs/az_paragraphs_photo_gallery/templates/paragraph--az-photo-gallery.html.twig [hunk 3]
```diff
@@ -57,21 +57,23 @@
         {% if content.field_az_photos[0] %}
         {% if grid %}
           {# Render the images for the gallery grid display. #}
```
Comment (trackleft) [inline modules/custom/az_paragraphs/az_paragraphs_photo_gallery/templates/paragraph--az-photo-gallery.html.twig:59]: Were these changes on purpose?

File: modules/custom/az_paragraphs/az_paragraphs.install (hunk 2 above)
Comment (trackleft) [reply modules/custom/az_paragraphs/az_paragraphs.install:18]: NVM I See you put 

> Added explicit access checks to entity queries for [drupal.org/node/3201242](https://www.drupal.org/node/3201242) (deprecation introduced in Drupal 9.2.x)

Comment (trackleft) [reply modules/custom/az_paragraphs/az_paragraphs.install:18]: NVM I see this now 

> Addressed Twig deprecations (see [twig.symfony.com/doc/2.x/deprecated.html#tags](https://twig.symfony.com/doc/2.x/deprecated.html#tags))
Adding an if condition on a for tag is deprecated in Twig 2.10. Use a filter filter or an "if" condition inside the "for" body instead (if your condition depends on a variable updated inside the loop).

File: modules/custom/az_paragraphs/az_paragraphs_photo_gallery/templates/paragraph--az-photo-gallery.html.twig (hunk 3 above)
Comment (trackleft) [reply modules/custom/az_paragraphs/az_paragraphs_photo_gallery/templates/paragraph--az-photo-gallery.html.twig:59]: NVM I see this now

> Addressed Twig deprecations (see [twig.symfony.com/doc/2.x/deprecated.html#tags](https://twig.symfony.com/doc/2.x/deprecated.html#tags))
Adding an if condition on a for tag is deprecated in Twig 2.10. Use a filter filter or an "if" condition inside the "for" body instead (if your condition depends on a variable updated inside the loop).

**Review by trackleft — DISMISSED** [review]

File: modules/custom/az_paragraphs/az_paragraphs.install (hunk 2 above)
Comment (joeparsons) [reply modules/custom/az_paragraphs/az_paragraphs.install:18]: Yes.  See the updated issue description.

File: modules/custom/az_paragraphs/az_paragraphs_photo_gallery/templates/paragraph--az-photo-gallery.html.twig (hunk 3 above)
Comment (joeparsons) [reply modules/custom/az_paragraphs/az_paragraphs_photo_gallery/templates/paragraph--az-photo-gallery.html.twig:59]: Yes.  See the updated issue description.

**Comment by joeparsons** [comment]
Updated with new 9.5.0-rc2 release:
https://www.drupal.org/project/drupal/releases/9.5.0-rc2

File: az_quickstart.install [hunk 4]
```diff
@@ -233,3 +233,16 @@ function az_quickstart_update_920103() {
 function az_quickstart_update_920601() {
   \Drupal::service('module_installer')->uninstall(['media_library_theme_reset']);
 }
+
+/**
+ * Uninstall deprecated Drupal core modules.
+ */
+function az_quickstart_update_920602() {
+  $deprecated_modules = [
+    'color',
```
Comment (joeparsons) [inline az_quickstart.install:242]:
```suggestion
    'aggregator',
    'color',
```

**Comment by trackleft** [comment]
I like the direction this PR is turning

File: .probo.yaml [hunk 5]
```diff
@@ -17,6 +17,7 @@ steps:
       - composer config use-github-api false
       - composer require --no-update drupal/core-recommended:* zaporylie/composer-drupal-optimizations:* az-digital/az_quickstart:dev-${BRANCH_NAME}
       - composer require --no-update --dev az-digital/az-quickstart-dev:dev-${DEV_PACKAGE_BRANCH}
+      - composer require --no-update --dev drupal/upgrade_status:*
```
Comment (trackleft) [inline .probo.yaml:20]: Remove in favor of https://github.com/az-digital/az-quickstart-dev/issues/61

File: .probo.yaml [hunk 6]
```diff
@@ -45,6 +46,7 @@ steps:
   - name: Post-Installation Steps
     plugin: Script
     script:
+      - $SRC_DIR/az-quickstart-scaffolding/vendor/bin/drush -y -n pm:install upgrade_status
```
Comment (trackleft) [inline .probo.yaml:49]: Remove in favor of https://github.com/az-digital/az-quickstart-dev/issues/61

File: .probo.yaml (hunk 5 above)
Comment (trackleft) [inline .probo.yaml:20]:
```suggestion
```

**Comment by trackleft** [comment]
Test config distro updates for block class specifically

File: .probo.yaml (hunk 6 above)
Comment (trackleft) [inline .probo.yaml:49]:
```suggestion
```

Comment (joeparsons) [reply .probo.yaml:49]: I think we still want the pm:install step.

Comment (trackleft) [reply .probo.yaml:48]: I have another PR for this, so we can separate this point of contention (adds 14 minutes to builds) out https://github.com/az-digital/az_quickstart/pull/2111

Comment (trackleft) [reply .probo.yaml:48]: However, that PR is not against this branch, so either way is fine with me

File: composer.json [hunk 7]
```diff
@@ -138,16 +138,15 @@
                 "Unnecessary restrictions on logo format: Can't upload replacement SVG logo (2259567)": "https://www.drupal.org/files/issues/2022-05-05/2259567-160-9-3-x.patch",
                 "Ajax CSS load order issue (1461322)": "https://www.drupal.org/files/issues/2020-06-05/1461322-25.patch",
                 "Layout builder revisions issue (3033516)": "https://www.drupal.org/files/issues/2019-06-17/3033516-17.patch",
-                "Allow text field to enforce a specific text format (784672)": "https://www.drupal.org/files/issues/2022-05-13/784672-231.patch",
+                "Allow text field to enforce a specific text format (784672)": "https://www.drupal.org/files/issues/2022-10-21/784672-271.patch",
```
Comment (trackleft) [inline composer.json:141]:
```suggestion
                "Allow text field to enforce a specific text format (784672)": "https://www.drupal.org/files/issues/2023-01-05/784672-302.patch",
```

**Review by trackleft — APPROVED** [review]

**Review by bberndt-uaz — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Review by camikazegreen — APPROVED** [review]
